dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -78,7 +78,6 @@ bool AggregatedDataVariants::tryMarkNeedSpill()
         /// Data can only be flushed to disk if a two-level aggregation is supported.
         if (!isConvertibleToTwoLevel())
             return false;
-        convertToTwoLevel();
```

Comment:
Why was it moved to the `spill` function? Isn't `tryMarkNeedSpill` being called in parallel?

Reply:
Not in parallel. `tryMarkNeedSpill` can be called in `AggregateContext::needSpill`.